src/lib.rs

```diff
@@ -103,31 +105,30 @@ macro_rules! assert_unique_used_features {
     }
 }
 
-assert_unique_used_features! {"llvm4-0", "llvm5-0", "llvm6-0", "llvm7-0", "llvm8-0", "llvm9-0", "llvm10-0", "llvm11-0", "llvm12-0", "llvm13-0", "llvm14-0"}
+assert_unique_used_features! {"llvm4-0", "llvm5-0", "llvm6-0", "llvm7-0", "llvm8-0", "llvm9-0", "llvm10-0", "llvm11-0", "llvm12-0", "llvm13-0", "llvm14-0", "llvm15-0"}
 
-/// Defines the address space in which a global will be inserted.
-///
-/// # Remarks
-/// See also: https://llvm.org/doxygen/NVPTXBaseInfo_8h_source.html
+/// Defines the abstract LLVM address space.
 #[derive(Debug, PartialEq, Eq, Copy, Clone)]
 pub enum AddressSpace {
-    Generic = 0,
-    Global = 1,
-    Shared = 3,
-    Const = 4,
-    Local = 5,
+    Zero = 0,
```

Comment:
could we implement `Default` for `AddressSpace` and have that use address space Zero? Especially if the annotation shows up in the docs (and I'd guess that it does), (see https://github.com/rust-lang/rfcs/pull/3107) then that is a nice bit of extra documentation if you're not sure which one to pick.

Reply:
ah, this may conflict with the msrv policy

Reply:
Actually we should remove this enum altogether, because it blocks some of the address space usage.
LLVM allows almost unbounded number of address spaces (up to 2^24 IIRC), so we must just use u32 instead.

Reply:
I just had a thought that it wasn't a change everybody would agree on easily, so I decided to do it in a separate patch later in order not to block this one.

Reply:
that would just mean everyone has to go over all usages twice. if we implement default now, and use that in most places, that minimizes breakage. (assuming the `0u32` is also a good default)

Reply:
Ok, I have nothing against that default as long as we have that enum.